iothub/device/src/Transport/RetryDelegatingHandler.cs

```diff
@@ -226,6 +227,90 @@ public override async Task<Message> ReceiveAsync(TimeoutHelper timeoutHelper)
             }
         }
 
+        public override async Task EnableReceiveMessageAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (Logging.IsEnabled)
+                {
+                    Logging.Enter(this, cancellationToken, nameof(EnableReceiveMessageAsync));
+                }
+
+                await _internalRetryPolicy
+                    .ExecuteAsync(
+                        async () =>
+                        {
+                            // Ensure that the connection has been opened, before enabling the callback for receiving messages.
+                            await EnsureOpenedAsync(cancellationToken).ConfigureAwait(false);
+
+                            // Wait to acquire the _handlerLock. This ensures that concurrently invoked API calls are invoked in a thread-safe manner.
+                            await _handlerLock.WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
I would rename _handlerLock to _handlerSemaphore to make it clear for the reader that this is a semaphore ... just like what you did in InternalClient

Reply:
sure, I can do that